File: 4135-concatenate-non-zero-digits-and-multiply-by-sum-i/concatenate-non-zero-digits-and-multiply-by-sum-i.java
class Solution {
    public long sumAndMultiply(int n) {
        if(n==0) return 0;

        long x = 0;
        long sum = 0;
        long placeholder = 1;

        while(n>0){
            long rem = n%10;
            if(rem!=0){
                x += rem*placeholder;
                sum += rem; 
                placeholder *= 10;
            }
            n/=10;
            
        }
        return sum*x;
    }
}

File: 4135-concatenate-non-zero-digits-and-multiply-by-sum-i/concatenate-non-zero-digits-and-multiply-by-sum-i.py
class Solution:
    def sumAndMultiply(self, n: int) -> int:
        if n<=0:
            return 0
        x = 0
        s = ""
        while n>0:
            rem = n%10
            x+=rem
            if rem!=0:
                s += str(rem)
            n//=10
        s=s[::-1]
        return int(s)*x